icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ULocaleTest.java

```diff
@@ -1227,14 +1247,44 @@ public Item(String dLoc, DisplayContext dia, DisplayContext cap, DisplayContext
                 DisplayContext substituteHandling = ldn.getContext(DisplayContext.Type.SUBSTITUTE_HANDLING);
                 if (dialectHandling != item.dialectHandling || capitalization != item.capitalization || nameLength != item.nameLength || substituteHandling != item.substituteHandling) {
                     errln("FAIL: displayLoc: " + item.displayLocale + ", dialectNam?: " + item.dialectHandling +
-                            ", capitalize: " + item.capitalization + ", nameLen: " + item.nameLength + ", substituteHandling: " + item.substituteHandling + ", locToName: " + item.localeToBeNamed +
-                            ", => read back dialectNam?: " + dialectHandling + ", capitalize: " + capitalization + ", nameLen: " + nameLength + ", substituteHandling: " + substituteHandling);
+                            ", capitalize: " + item.capitalization + ", nameLen: " + item.nameLength +
+                            ", substituteHandling: " + item.substituteHandling + ", locToName: " + item.localeToBeNamed +
+                            ", => read back dialectNam?: " + dialectHandling + ", capitalize: " + capitalization +
+                            ", nameLen: " + nameLength + ", substituteHandling: " + substituteHandling);
                 } else {
+                    boolean checkULocaleDisplay = (dialectHandling==NM_STD && capitalization==CAP_MID && nameLength==LEN_FU && substituteHandling==SUB_SU);
+                    boolean checkJustLanguage = item.localeToBeNamed.matches("[a-z]{2,3}");
```

Comment:
Below, to the existing tests for LocaleDisplayNames.localeDisplayName, we add tests where relevant for ULocale.getDisplayName, LocaleDisplayNames.languageDisplayName, and ULocale.getDisplayLanguage.